module-4/backend/week_one.md

```diff
@@ -0,0 +1,32 @@
+## Week One - Module 4 Recap
+
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+
+Note: When you're done, submit a PR. 
+
+1. What's the most useful thing you learned from completing the intermission week work?
+It was all useful, i can't think of one specific thing.
+2. What are some tools to help debug JavaScript code?
+pry and console.log()
+3. What are some tools you need in order to unit test your JavaScript?
+mocha / chai
+4. What is the syntax for invoking a function?
+function()
+5. What's the difference between `==` and `===` in JavaScript?
+'==' is converts both values to the same type and then compares, '===' does not do any type conversion
+6. What's the difference between asynchronous and synchronous JavaScript? 
+synchronous javascript runs sequentially and asynchronous calls a function before the previous function is finished
+7. How do we setup a route when creating an API with Node and Express?
+in the server file app.get or app.post then the route
+8. What do we use Knex for?
+Running/Controlling migrations
+9. What's `npm` and what do we use it for?
+node package manager, it's like a gemfile
+
+#### Review  
+10. What's the MVC design pattern? Describe each part of MVC?
+Model View Controller, splitting things up based on their functionality
+11. What is AJAX? What are some benefits of using AJAX?
+ajax can make requests to the server without reloading the page
+12. What's a background worker? When would we want to use a background worker?
```

Comment:
@seth-at-at please review the comments. I'd like to see more detailed responses in the future.